src/DynamoPublish/DynamoPublishExtension.cs

```diff
@@ -111,29 +118,112 @@ private MenuItem GenerateMenuItem()
+        private void TermsOfUseEnabled()
+        {
+            Task<bool>.Factory.StartNew(() => shareWorkspaceClient.GetTermsOfUseAcceptanceStatus()).
+                ContinueWith(t =>
                 {
-                    Owner = loadedParams.DynamoWindow,
-                    WindowStartupLocation = WindowStartupLocation.CenterOwner
-                };
+                    // The above GetTermsOfUseAcceptanceStatus call will get the
+                    // user to sign-in. If the user cancels that sign-in dialog 
+                    // without signing in, we won't show the terms of use dialog,
+                    // simply return from here.
+                    // 
+                    if (startupParams.AuthProvider.LoginState != LoginState.LoggedIn)
+                        return;
+
+                    var termsOfUseAccepted = t.Result;
+                    if (termsOfUseAccepted)
+                    {
+                        // Terms of use accepted, proceed to publish.
+                        MessageBoxResult decision =
+                        MessageBox.Show(Resources.TermsConfirmationContent,
+                            Resources.TermsConfirmation,
+                            MessageBoxButton.YesNo, MessageBoxImage.Question);
+                        if (decision == MessageBoxResult.Yes)
+                        {
+                            ProcessPublish();
+                        }
+                    }
+                    else
+                    {
+                        // Prompt user to accept the terms of use.
+                        ShowTermsOfUseForPublishing();
+                    }
+
+                }, TaskScheduler.FromCurrentSynchronizationContext());
+        }
 
-                window.ShowDialog();
+        private void ProcessPublish()
+        {
+            var model = new PublishModel(startupParams.AuthProvider, startupParams.CustomNodeManager);
+            model.MessageLogged += this.OnMessageLogged;
 
-                model.MessageLogged -= this.OnMessageLogged;
+            var viewModel = new PublishViewModel(model)
+            {
+                CurrentWorkspaceModel = loadedParams.CurrentWorkspaceModel,
+                Cameras = ConvertCameraData(loadedParams.BackgroundPreviewViewModel.GetCameraInformation())
             };
 
-            return item;
+            var window = new PublishView(viewModel)
+            {
+                Owner = loadedParams.DynamoWindow,
+                WindowStartupLocation = WindowStartupLocation.CenterOwner
+            };
+
+            window.ShowDialog();
+
+            model.MessageLogged -= this.OnMessageLogged;
+        }
+
+        internal static bool ShowTermsOfUseDialog()
+        {
+            var executingAssemblyPathName = Assembly.GetExecutingAssembly().Location;
+            var rootModuleDirectory = Path.GetDirectoryName(executingAssemblyPathName);
+            var touFilePath = Path.Combine(rootModuleDirectory, "TermsOfUse.rtf");
```

Comment:
Hi @pboyer, the document resides under `C:\Program Files\Dynamo 0.9` folder (the same folder that executing assembly `DynamoPublish.dll` resides in) and it is not localized, so this is good.